### DISCUSSION TITLE
Modular Development & Debugging source code

### DISCUSSION THREAD

**Discussion by Sharafudeen** [oqtane/oqtane.framework#2119] [General]
Hi,

I would like to setup development environment as follows,

1. **Developer-1:** Can develop Modules and customize Oqtane framework source code
2. **Developer-2:** Can develop Module1
3. **Developer-3:** Can develop Module2

**Conditions:**

**Developer-1** having full rights to edit Oqtane framework source.

**Developer-2** cannot edit Oqtane framework source but able to build Module1.

**Developer-3** cannot edit Oqtane framework source but able to build Module2.

How can Developer-2 & Developer-3 debug their modules without having Oqtane framework source code?

**Comment by sbwalker** [comment]
We do not recommend customizing the Oqtane framework source code, as if you go down that path, you will not be able to easily upgrade to newer versions of the framework without a lot of manual effort. That being said, it is sometimes helpful to be able to debug into the Oqtane framework source code in order to be able to understand the implementation details of how it works. But if you truly want to maintain a clear separation from the framework then you should use the official Installation/Upgrade release packages which only contain the framework binaries.

**Reply by Sharafudeen** [reply]
How can I run & debug standalone Oqtane module? Example, if I develop Customer module and I want to run & debug it's source code from Visual Studio 2022?

**Reply by leigh-pointer** [reply]
@Sharafudeen If you use the Create Module feature in the framework to scaffold the module code, then the solution will be set up and will allow you to debug you code.
